tinygrad/codegen/cstyle.py

```diff
@@ -46,6 +47,29 @@ def render_cast(self, x:List[str], var_dtype:DType) -> str:
     if var_dtype == dtypes._float2: return f"{self.float4.replace('float4', 'float2')}({','.join(x)})"
     raise NotImplementedError(f"no cast for {var_dtype}")
 
+  def render_bitcast(self, x:str, var_dtype:DType) -> str:
+    if self.is_gpu:
+      if var_dtype == dtypes.float32:
+        return f"as_float({x})"
+      if var_dtype == dtypes.float16:
+        return f"as_half({x})"
+      if var_dtype == dtypes.bfloat16:
+        raise NotImplementedError("No bitcast implemented for bfloat16 in OpenCL")
+      if var_dtype == dtypes.int8:
+        return f"as_char({x})"
+      if var_dtype == dtypes.int32:
+        return f"as_int({x})"
+      if var_dtype == dtypes.int64:
+        return f"as_long({x})"
+      if var_dtype == dtypes.uint8:
+        return f"as_uchar({x})"
+      if var_dtype == dtypes.uint32:
+        return f"as_uint({x})"
+      if var_dtype == dtypes.uint64:
+        return f"as_ulong({x})"
+      raise NotImplementedError(f"No bitcast implemented for {var_dtype}")
```

Comment:
There has to be a better way to do this.

Reply:
Refactored it